src/envs/cover.py

```diff
@@ -401,13 +405,27 @@ class CoverEnvRegrasp(CoverEnv):
     This environment also has two different oracle NSRTs for placing, one for
     placing a target and one for placing on the table.
 
+    This environment also has a Clear predicate, to prevent placing on already
+    covered targets.
+
     Finally, to allow placing on the table, we need to change the allowed
     hand regions. We implement it so that there is a relatively small hand
     region centered at each target, but then everywhere else is allowed.
     """
     _allow_free_space_placing = True
     _initial_pick_offsets = [-0.95, 0.0, 0.95]
 
+    def __init__(self) -> None:
+        super().__init__()
+        # Add a Clear predicate to prevent attempts at placing on already
```

Comment:
That's interesting. The lack of a Clear predicate was the reason why the base Cover env is not downward refinable right? (In combination with the fact that initial_holding_prob > 0).

Reply:
hmm you're right that the base Cover env is not downward refinable, in that there are skeletons that able to be refinable but are not. however, unlike in CoverRegrasp, we would never actually try to refine a nonrefinable skeleton in the base env, because all of those skeletons would just place the respective object onto the respect target immediately.

Reply:
note: we realized on slack that ^ is not true. suppose your goal is just b0 on t0, and you start out holding b1. a nonrefinable skeleton is [place b1 on t0, place b0 on t0]. so, base Cover env is not downward refinable!